### DISCUSSION TITLE
Problem with onRecordsListRequest pb_hook

### DISCUSSION THREAD

**Discussion by pietru2004** [pocketbase/pocketbase#6014] [Q&A] [answered]
Pocketbase Version: [0.22.27](https://github.com/pocketbase/pocketbase/releases/tag/v0.22.27)

Hello I want to ask should onRecordsListRequest event have `e.auth` as null/undefined when doing refreshing via admin dashboard

in attached script I fought it might be better to use `onRecordsListRequest` rather than `routerAdd`

[messages.pb.js.txt](https://github.com/user-attachments/files/17996340/messages.pb.js.txt)

It is related to https://github.com/pocketbase/pocketbase/discussions/5901, the pb schema is the same.

**Comment by ganigeorgiev** [comment, marked as answer]
`e.auth` is PocketBase v0.23+ only field.

For the PocketBase <= v0.22.x documentation you can check https://pocketbase.io/old/docs/.

Specifically for the auth access, see https://pocketbase.io/old/docs/js-routing/#retrieving-the-current-auth-state (_`c` in the example is `e.httpContext` from the hook_).

**Reply by pietru2004** [reply]
hm... I guess I should upgrade than

**Reply by pietru2004** [reply]
thx for help
